src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupBasedAlleles.java

```diff
@@ -1,25 +1,48 @@
 package org.broadinstitute.hellbender.utils.pileup;
 
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.util.SequenceUtil;
 import htsjdk.variant.variantcontext.Allele;
 import htsjdk.variant.variantcontext.VariantContext;
 import htsjdk.variant.variantcontext.VariantContextBuilder;
 import org.broadinstitute.hellbender.engine.AlignmentAndReferenceContext;
 import org.broadinstitute.hellbender.engine.AlignmentContext;
 import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.PileupDetectionArgumentCollection;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
 
 import java.util.*;
 
 
+/**
+ * Helper class for handling pileup allele detection supplement for assembly. This code is analogous but not exactly
+ * equivalent to the DRAGEN ColumnwiseDetection approach.
+ */
 public final class PileupBasedAlleles {
 
-
-//    ArrayList<VariantContext> forcedPileupAlleles;
-
-
-    public static ArrayList<VariantContext> getPileupVariantContexts(List<AlignmentAndReferenceContext> alignmentAndReferenceContextList) {
+    /**
+     * Accepts the raw per-base pileups stored from the active region detection code and parses them for potential variants
+     * that are visible in the pileups but might be dropped from assembly for any number of reasons. The basic algorithm works
+     * as follows:
+     *  - iterate over every pileup and count alt bases
+     *      - (unfinished) detect insertions overlapping this site (CURRENTLY ONLY WORKS FOR INSERTIONS)
```

Comment:
it's unclear what is unfinished...do you want this to work for indels, but it only works for insertions?

Reply:
I consider the implementation of indels to be relatively untested compared to the SNP code. I'm going to change this to beta.